web/packages/teleport/src/services/integrations/types.ts

```diff
@@ -33,18 +33,27 @@ import { Node } from '../nodes';
  * while "plugin" resource is only supported in enterprise. Plugin
  * type exists in OS for easier typing when combining the resources
  * into one list.
+ *
+ * Generics:
+ *  T is resource type "integration" or "plugin"
+ *  K is the kind of integration (eg: aws-oidc) or plugin (eg: okta)
+ *  SP is the provider-specific spec of integration or plugin
+ *  SD is the provider-specific status containing status details
+ *   - currently only defined for plugin resource
```

Comment:
Nit, but you can rename generics to `ResourceType`, `IntegrationKind`, etc., they don't have to be single or two letter names. 😏